packages/services/api/src/modules/oidc-integrations/providers/oidc-integrations.provider.ts

```diff
@@ -282,6 +282,36 @@ export class OIDCIntegrationsProvider {
     } as const;
   }
 
+  async updateOIDCRestrictions(args: { oidcIntegrationId: string; oidcUserAccessOnly: boolean }) {
+    if (this.isEnabled() === false) {
+      return {
+        type: 'error',
+        message: 'OIDC integrations are disabled.',
+      } as const;
+    }
+
+    const oidcIntegration = await this.storage.getOIDCIntegrationById({
+      oidcIntegrationId: args.oidcIntegrationId,
+    });
+
+    if (oidcIntegration === null) {
+      return {
+        type: 'error',
+        message: 'Integration not found.',
+      } as const;
+    }
+
+    await this.authManager.ensureOrganizationAccess({
+      organization: oidcIntegration.linkedOrganizationId,
+      scope: OrganizationAccessScope.INTEGRATIONS,
```

Comment:
should we also limit it to `SETTINGS`? or `INTEGRATIONS` is specific enough? 
I imagine only admin should have permissions to change that?

Reply:
`organization:integrations` is what @n1ru4l used for the entire OIDC setup, so I followed